File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/ui/utility/MentionList.java
package io.getstream.chat.docs.java.ui.utility;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import io.getstream.chat.android.ui.feature.mentions.list.MentionListView;
import io.getstream.chat.android.ui.viewmodel.mentions.MentionListViewModel;
import io.getstream.chat.android.ui.viewmodel.mentions.MentionListViewModelBinding;
import io.getstream.chat.android.ui.viewmodel.mentions.MentionListViewModelFactory;

/**
 * [Mention List View](https://getstream.io/chat/docs/sdk/android/ui/utility-components/mention-list-view/)
 */
public class MentionList extends Fragment {

    private MentionListView mentionListView;

    /**
     * [Usage](https://getstream.io/chat/docs/sdk/android/ui/utility-components/mention-list-view/#usage)
     */
    public void usage() {
        ViewModelProvider.Factory factory = new MentionListViewModelFactory();
        MentionListViewModel viewModel = new ViewModelProvider(this, factory).get(MentionListViewModel.class);
        MentionListViewModelBinding.bind(viewModel, mentionListView, getViewLifecycleOwner());
    }

    /**
     * [Handling Actions](https://getstream.io/chat/docs/sdk/android/ui/utility-components/mention-list-view/#handling-actions)
     */
    public void handlingActions() {
        mentionListView.setMentionSelectedListener(message -> {
            // Handle a mention item being clicked
        });
    }
}

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/ui/utility/MentionList.kt
package io.getstream.chat.docs.kotlin.ui.utility

import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import io.getstream.chat.android.ui.feature.mentions.list.MentionListView
import io.getstream.chat.android.ui.viewmodel.mentions.MentionListViewModel
import io.getstream.chat.android.ui.viewmodel.mentions.MentionListViewModelFactory
import io.getstream.chat.android.ui.viewmodel.mentions.bindView

/**
 * [Mention List View](https://getstream.io/chat/docs/sdk/android/ui/utility-components/mention-list-view/)
 */
class MentionList : Fragment() {

    private lateinit var mentionListView: MentionListView

    /**
     * [Usage](https://getstream.io/chat/docs/sdk/android/ui/utility-components/mention-list-view/#usage)
     */
    fun usage() {
        val viewModel: MentionListViewModel by viewModels { MentionListViewModelFactory() }
        viewModel.bindView(mentionListView, viewLifecycleOwner)
    }

    /**
     * [Handling Actions](https://getstream.io/chat/docs/sdk/android/ui/utility-components/mention-list-view/#handling-actions)
     */
    fun handlingActions() {
        mentionListView.setMentionSelectedListener { message ->
            // Handle a mention item being clicked
        }
    }
}
